src/AzureExtension/DevBox/Models/CreateComputeSystemOperation.cs

```diff
@@ -74,7 +105,7 @@ public void CompleteWithFailure(Exception exception, string displayText)
         }
 
         IsCompleted = true;
-        Completed?.Invoke(this, new CreateComputeSystemResult(exception, displayText, exception.Message));
+        Result ??= new CreateComputeSystemResult(exception, displayText, exception.Message);
```

Comment:
In what case Result can be not null? This is confusing since we are saying "complete operation with this error", but then ignoring the error. If operation is already completed and we are somehow completing it second time we should return under "if (IsCompleted)" (if a race is possible here then we need to use locking here).

Reply:
thanks I updated this, no more  CompletedWith... methods. Now we get the CreateCumpteSystemResult directly from the StartCreateDevBoxOperation method in the DevBoxCreationManager object. Then return that. If a second one comes in we return the same result object.